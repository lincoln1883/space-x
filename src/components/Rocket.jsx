import React from 'react';
import PropTypes from 'prop-types';

const Rocket = ({
  name, image, description,
}) => (
  <>
    <li className="d-flex gap-3 mb-4">
      <img width="200" height="200" src={image} alt={name} />
      <div className="flex flex-col gap-3">
        <h3>{name}</h3>
        <p>{description}</p>
        <button className="btn btn-primary" type="button">Reserve Rocket</button>
      </div>
    </li>
  </>
);

Rocket.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};
export default Rocket;
