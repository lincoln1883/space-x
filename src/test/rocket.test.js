import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import mockAxios from 'axios';
import store from '../redux/store';
import Rocket from '../components/Rocket';

describe('Rocket', () => {
  describe('Rocket component', () => {
    it('renders a list of rockets', () => {
      mockAxios.get.mockResolvedValue({
        data: [
          {
            id: '5e9d0d95eda69973a809d1ec',
            name: 'Falcon 1',
            description:
              'The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.',
            images: ['https://i.imgur.com/4Xf8rvI.jpg'],
          },
        ],
      });
      const rocket = render(
        <Provider store={store}>
          <Rocket />
        </Provider>,
      );
      expect(rocket).toMatchSnapshot();
    });
    it('should render rocket component', () => {
      const tree = render(
        <Provider store={store}>
          <Rocket />
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
    });
  });
});
