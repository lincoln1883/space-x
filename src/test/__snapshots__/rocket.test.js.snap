// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket Rocket component renders a list of rockets 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="d-flex gap-3 mb-4"
      >
        <img
          height="200"
          width="200"
        />
        <div
          class="flex flex-col gap-3"
        >
          <h3 />
          <p />
          <button
            class="btn btn-primary"
            type="button"
          >
            Reserve Rocket
          </button>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="d-flex gap-3 mb-4"
    >
      <img
        height="200"
        width="200"
      />
      <div
        class="flex flex-col gap-3"
      >
        <h3 />
        <p />
        <button
          class="btn btn-primary"
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rocket Rocket component should render rocket component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="d-flex gap-3 mb-4"
      >
        <img
          height="200"
          width="200"
        />
        <div
          class="flex flex-col gap-3"
        >
          <h3 />
          <p />
          <button
            class="btn btn-primary"
            type="button"
          >
            Reserve Rocket
          </button>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="d-flex gap-3 mb-4"
    >
      <img
        height="200"
        width="200"
      />
      <div
        class="flex flex-col gap-3"
      >
        <h3 />
        <p />
        <button
          class="btn btn-primary"
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
